test/rest/http.test.js

```diff
@@ -62,5 +62,26 @@ define(['ably', 'shared_helper', 'chai'], function (Ably, helper, chai) {
 
       done();
     });
+
+    it('Should handle no content responses', function (done) {
+      //Intercept Http.do with test
+
+      function testRequestHandler(_, __, ___, ____, _____, ______, callback) {
```

Comment:
not really important but might slightly simpler to test this by using `https://echo.ably.io/respondWith?status=204`

Reply:
I did try this but sadly it didn't like it :(